remote-api/src/main/java/org/alfresco/rest/api/model/rules/Rule.java

```diff
@@ -45,13 +44,28 @@ public static Rule from(final org.alfresco.service.cmr.rule.Rule ruleModel) {
             return null;
         }
 
-        final Rule rule = new Rule();
+        final Rule rule = new RuleBuilder().createRule();
         rule.id = ruleModel.getNodeRef().getId();
         rule.name = ruleModel.getTitle();
 
         return rule;
     }
 
+    /**
+     * Convert the REST model object to the equivalent service POJO.
+     *
+     * @param nodes The nodes API.
+     * @return The rule service POJO.
+     */
+    public org.alfresco.service.cmr.rule.Rule toServiceModel(Nodes nodes)
```

Comment:
It's a bit awkward to do this, because then whenever we want to call `rule.toServiceModel` then we have to do:
```
NodeRef ruleNode = nodes.validateOrLookupNode(rule.getId(), null);
rule.toServiceModel(ruleNode);
```
compared with:
```
rule.toServiceModel(nodes);
```
I agree that the current logic is twisted though - I just couldn't find a nicer way than this.

Reply:
As said before - I don't mind when it stays as is.